### PR TITLE
feat: support flux type in dynamic

### PR DESCRIPTION
**Pull request by DanLi39**

### CODE REVIEW

File: arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/ReactorStreamConsumer.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    /**
+     * support for Mono type recording
+     * @param responseMono
+     * @return
+     */
+    public Mono<?> monoAccept(Mono<?> responseMono) {
+        return responseMono
+            .doOnSuccess(result -> {
+                traceTransmitter.transmit();
```
Comment (YongwuHe) [inline arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/ReactorStreamConsumer.java:29]: try(traceTransmitter.transmit()), need close

File: arex-instrumentation/dynamic/arex-dynamic-common/src/test/java/io/arex/inst/dynamic/common/DynamicClassExtractorTest.java [hunk 2]
```diff
@@ -484,6 +506,13 @@ public Mono<String> testReturnMono(String val, Throwable t) {
         return Mono.justOrEmpty(val + "testReturnMono");
     }
 
+    public Flux<String> testReturnFlux(String val,Throwable t){
```
Comment (mr3) [inline arex-instrumentation/dynamic/arex-dynamic-common/src/test/java/io/arex/inst/dynamic/common/DynamicClassExtractorTest.java:509]: `testReturnFlux(String val,Throwable t)`

add space after `,`

File: arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/DynamicClassExtractor.java [hunk 3]
```diff
@@ -308,6 +315,12 @@ Object restoreResponse(Object result) {
             return Mono.justOrEmpty(result);
         }
 
+        if (FLUX.equals(this.methodReturnType)) {
+            if (result instanceof Throwable) {
+                return Flux.error((Throwable) result);
+            }
+            return result == null ? Flux.empty() : Flux.just(result);
```
Comment (mr3) [inline arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/DynamicClassExtractor.java:322]: `Flux.just(List)` is different from `Flux.fromArray/fromIterable`

File: arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/ReactorStreamConsumer.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            })
+            .doFinally(result -> {
+                traceTransmitter.transmit();
+                if (doOnError.get()) {
+                    extractor.recordResponse(error[0]);
+                } else {
+                    extractor.recordResponse(fluxList.toArray());
+                }
+            });
+    }
```
Comment (mr3) [inline arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/ReactorStreamConsumer.java:64]: A Reactive Streams [Publisher](https://www.reactive-streams.org/reactive-streams-1.0.3-javadoc/org/reactivestreams/Publisher.html?is-external=true) with rx operators that emits 0 to N elements, and then completes (successfully or with an error).

File: arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/MonoConsumer.java [hunk 5]
```diff
@@ -14,17 +14,23 @@ public MonoConsumer(DynamicClassExtractor extractor) {
...
-                }
+            .doOnSuccess(result -> {
+                traceTransmitter.transmit();
+                extractor.recordResponse(result);
+                traceTransmitter.close();
             })
-            .doOnError(o -> {
-                try (TraceTransmitter tm = traceTransmitter.transmit()) {
-                    extractor.recordResponse(o);
-                }
```
Comment (mr3) [inline arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/MonoConsumer.java:27]: see [The try-with-resources Statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)

File: arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/DynamicClassExtractor.java [hunk 6]
```diff
@@ -308,6 +316,12 @@ Object restoreResponse(Object result) {
             return Mono.justOrEmpty(result);
         }
 
+        if (FLUX.equals(this.methodReturnType)) {
+            if (result instanceof Throwable) {
+                return Flux.error((Throwable) result);
+            }
+            return result == null ? Flux.empty() : FluxConsumer.restore((result));
```
Comment (mr3) [inline arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/DynamicClassExtractor.java:323]: Just `return FluxConsumer.restore((result));`, the rest of the logic can be wrapped in `restore` method

File: arex-instrumentation/dynamic/arex-dynamic-common/src/test/java/io/arex/inst/dynamic/common/listener/FluxConsumerTest.java [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+        //testFluxFromArray
+        Flux<?> testFluxFromArray = testFluxFromArray();
+        Flux finalTestFluxFromArray = fluxConsumer.accept(testFluxFromArray);
+        assertEquals(1, finalTestFluxFromArray.blockFirst());
+
+        //testFluxFromStream
+        Flux<?> testFluxFromStream = testFluxFromStream();
+        Flux finalTestFluxFromStream = fluxConsumer.accept(testFluxFromStream);
+        assertEquals(1, finalTestFluxFromStream.blockFirst());
```
Comment (mr3) [inline arex-instrumentation/dynamic/arex-dynamic-common/src/test/java/io/arex/inst/dynamic/common/listener/FluxConsumerTest.java:96]: Recommend naming test methods base on business scenarios

- testAllElementNotNull
- testElementContainsNullOrExpcetion
- testEmptyElement

File: arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/FluxConsumer.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+        this.extractor = extractor;
+    }
+
+    public Flux<?> accept(Flux<?> responseFlux) {
+        List<FluxElementMockerResult> fluxElementMockerResults = new ArrayList<>();
+        AtomicInteger index = new AtomicInteger(1);
+        String responseType = TypeUtil.getName(responseFlux);
+        return responseFlux
+            .doOnNext(element ->
+                fluxElementMockerResults.add(
```
Comment (lucas-myx) [inline arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/FluxConsumer.java:37]: `traceTransmitter.transmit();` prevent thread change

File: arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/FluxConsumer.java [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+                    getFluxElementMockerResult(index.getAndIncrement(), element)))
+            .doOnError(error -> fluxElementMockerResults.add(
+                getFluxElementMockerResult(index.getAndIncrement(), error)))
```
Comment (lucas-myx) [inline arex-instrumentation/dynamic/arex-dynamic-common/src/main/java/io/arex/inst/dynamic/common/listener/FluxConsumer.java:40]: `traceTransmitter.transmit();` prevent thread change

File: arex-instrumentation/common/arex-common/src/main/java/io/arex/inst/common/util/FluxUtil.java [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    static final String FLUX_FROM_STREAM = "reactor.core.publisher.FluxStream-";
+
+    private FluxUtil() {
+    }
+
+    public static Flux<?> restore(Object fluxResult) {
+        if(fluxResult == null){
+            return Flux.empty();
+        }
+        List<FluxElementResult> result = ((FluxResult) fluxResult).getFluxElementResults();
```
Comment (lucas-myx) [inline arex-instrumentation/common/arex-common/src/main/java/io/arex/inst/common/util/FluxUtil.java:26]:
```
FluxResult fluxResult = (FluxResult) fluxObj;
List<FluxElementResult> result = fluxResult.getFluxElementResults();
String responseType = fluxResult.getResponseType();
```

File: arex-instrumentation/common/arex-common/src/main/java/io/arex/inst/common/util/FluxUtil.java [hunk 11]
```diff
@@ -0,0 +1,97 @@
+package io.arex.inst.common.util;
+
+import io.arex.agent.bootstrap.util.StringUtil;
+import io.arex.inst.runtime.serializer.Serializer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import reactor.core.publisher.Flux;
+import org.springframework.util.CollectionUtils;
```
Comment (lucas-myx) [inline arex-instrumentation/common/arex-common/src/main/java/io/arex/inst/common/util/FluxUtil.java:10]: io.arex.agent.bootstrap.util.CollectionUtil

**Review by lucas-myx — APPROVED** [review]

File: arex-instrumentation/common/arex-common/pom.xml [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>arex-common</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>io.projectreactor</groupId>
+            <artifactId>reactor-core</artifactId>
+            <version>${reactor.version}</version>
+            <scope>compile</scope>
```
Comment (YongwuHe) [inline arex-instrumentation/common/arex-common/pom.xml:20]: provided

**Review by lucas-myx — APPROVED** [review]

**Review by YongwuHe — APPROVED** [review]

**Pull request merged**